polaris/polaris/helpers.py

```diff
@@ -120,3 +126,95 @@ def validate_jwt_request(request: Request) -> str:
         return jwt_dict["sub"]
     except KeyError:
         raise ValueError("Decoded JWT missing 'sub' field")
+
+
+def validate_interactive_jwt(content_type: str = "text/html") -> Callable:
+    """
+    Decorator to validate the JWT generated by /interactive endpoints
+    for /webapp endpoints.
+    """
+
+    def decorator(view) -> Callable:
+        def wrapper(request: Request, *args, **kwargs) -> Response:
+            try:
+                if request.method == "GET":
```

Comment:
It seems odd to determine the action by the request method.  Would it make sense to have a separate decorator for the first case `authenticate_interactive_jwt` that converts the jwt to a session, and then `authenticate_interactive_session` for actually checking the session authentication?  It would be clearer in the code (`validate_interactive_jwt` isn't actually whats happening for either case) and would also allow us to not have a problem if we need to have any GET requests in the flow.

Reply:
I like that solution, the only issue is that we'd have 2 decorators on the same function, and only 1 of them would take affect, while the other one would have to simply pass the view function along.

If we pair this change with your suggestion to split up the `GET` and `POST` requests into separate endpoints, having two decorators would make a lot more sense.

Reply:
I renamed the decorator to `@interactive_authentication` and filed an issue to split the interactive endpoints by method.